tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/DataStreamsMonitoringRabbitMQTests.cs

```diff
@@ -0,0 +1,97 @@
+// <copyright file="DataStreamsMonitoringRabbitMQTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.TestHelpers;
+using Datadog.Trace.TestHelpers.DataStreamsMonitoring;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using ICSharpCode.Decompiler.Util;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests;
+
+[UsesVerify]
+[Trait("RequiresDockerDependency", "true")]
+public class DataStreamsMonitoringRabbitMQTests : TestHelper
+{
+    public DataStreamsMonitoringRabbitMQTests(ITestOutputHelper output)
+        : base("DataStreams.RabbitMQ", output)
+    {
+        SetServiceVersion("1.0.0");
+        EnableDebugMode();
+    }
+
+    [SkippableFact]
+    [Trait("Category", "EndToEnd")]
+    public async Task HandleProduceAndConsume()
+    {
+        SetEnvironmentVariable(ConfigurationKeys.DataStreamsMonitoring.Enabled, "1");
+
+        using var assertionScope = new AssertionScope();
+        using var agent = EnvironmentHelper.GetMockAgent();
+        using (RunSampleAndWaitForExit(agent, arguments: $"{TestPrefix}"))
+        {
+            var payloads = agent.WaitForDataStreams(1);
```

Comment:
Due to the way flushing works, this has the _potential_ to be flaky, as we _might_ not send everything in a single payload. I'm also concerned about how stable the payload is for snapshots (for the kafka tests it was _very_ unstable and caused a lot of flakiness).

I would suggest doing the following:
- Renaming the _existing_ `DataStreamsMonitoringTests` (which are badly named now, as they are Kafka only),  to `DataStreamsMonitoringKafkaTests` (or similar)
- Creating a `DataStreamsMonitoringTests` abstract base class in `Datadog.Trace.ClrProfiler.IntegrationTests/Helpers`.
- Move the `NormalizeDataStreams`, `StableSort`, `GetRootHashName`, `GetHashDepth` and `ScrubByteArray` methods to the base class
- Use those methods in the new rabbitmq tests to try to avoid issues with flakiness